### ISSUE TITLE
Go to definition doesn't work for vendored libraries

### ISSUE THREAD

**Issue by volkanunsal** [Microsoft/vscode-go#320]
The layout of my project is as follows:

```
- src
  - projectname 
    - main.go
- vendor
  - src
    - github.com
      - user
         - library
            - main.go
```

Currently, I cannot jump to the definition of the symbols defined in a vendored library. I wonder if this is due to the way my project is set up and can be fixed by setting a shell variable somehow or is it something that needs to be solved by this library? Thanks in advance.

**Comment by lukehoban** [comment]
This is a limitation of `godef`.  See https://github.com/rogpeppe/godef/pull/25 and https://github.com/rogpeppe/godef/issues/13.  Sounds like this is just waiting on @rogpeppe to review the open PR there.

**Comment by tilinna** [comment]
Looks that @rogpeppe is no longer maintaining godef. Have you considered other options?

**Comment by tilinna** [comment]
The issue is now fixed on godef master, thanks @rogpeppe :)

**Comment by volkanunsal** [comment]
How can I activate this feature in my project? I upgraded both godef and vscode-go, but it still complains about not finding the packages in vendor directory.

**Comment by tilinna** [comment]
Perhaps you forgot to use the -u flag on go get?

`go get -u github.com/rogpeppe/godef`

**Comment by volkanunsal** [comment]
I'm sure I did use the -u flag in combination with the -v flag. That should be ok, right?

Here is a screenshot of my workspace. You can see the squiggly line under the import statement, even though the package is in the vendor directory. 

![screen shot 2016-05-30 at 11 33 58 am](https://cloud.githubusercontent.com/assets/151600/15653367/a164c6b6-265a-11e6-813e-8516a2ab266b.png)

**Comment by tilinna** [comment]
You are right, it doesn't function properly. I think it "worked" for me because I had some leftovers around. There seems to be yet another issue: rogpeppe/godef#39

**Comment by rogpeppe** [comment]
The directory hierarchy shown in that image shows the "vendor" directory underneath
the very top level of $GOPATH, which I don't believe is supported.

Specifically, it shows:

    /src/sqlserver/fixtures/main.go
    /vendor/src/github.com/jmoiron/sqlx

but vendor directories must be inside the source directory, as documented
in [go/build](https://golang.org/pkg/go/build/#ImportMode):

> Import searches vendor directories that apply in the given source directory

To make this work, you'll want to put the sqlx package
inside /src/vendor.

**Comment by volkanunsal** [comment]
@rogpeppe I tried that and didn't work. Also, when I hover over the squigly line it tells me which directories the extension is looking through. There are two directories on the list and vendor directory is not on it.

**Comment by rogpeppe** [comment]
@volkanunsal Could you provide a tar archive that demonstrates the issue? I can't reproduce your situation, so I need more information if I'm to fix your problem. Thanks.

**Comment by rogpeppe** [comment]
For the record, I just created the following files:

	# /tmp/v/src/vendor/github.com/jmoiron/sqlx/sqlx.go
	package sqlx
	var X = 34

	# /tmp/v/src/sql-server/fixtures/main.go
	package main
	import (
		"github.com/jmoiron/sqlx"
	)
	var x = sqlx.X
	func main() {}

I then set GOPATH to /tmp/v, and godef
found github.com/jmoiron/sqlx from src/sql-server/fixtures/main.go
just fine (it finds sqlx.X too).

**Comment by volkanunsal** [comment]
@rogpeppe Ah, I believe the issue is I'm not setting GOPATH to my project directory. My project is laid out using the vendor convention of a [gb project](https://getgb.io/examples/sample-project/). I suppose then I'm out of luck unless I move my vendor directory under src folder?

**Comment by rogpeppe** [comment]
Yes, that would explain things. Godef supports the standard Go conventions. It may work if you set GOPATH="$PROJECT:$PROJECT/vendor" where $ PROJECT is your project root directory.

**Comment by wzzhu** [comment]
Setting GOPATH="$PROJECT:$PROJECT/vendor" won't work as godef will look for $GOPATH/src, while vendor has no intermediate "src" dir, e.g., vendor/github.com/...

On 1.7.1, macOS Sierra, the problem appears again. godef won't search vendor dir even if the layout is src/vendor/github.com/... Please take a look.
